### PR TITLE
Add extra Salt Bundle details

### PR DESCRIPTION
**Pull request by vzhestkov**
# Description

Enhances Salt Bundle documentation.

# Target branches

Which documentation version does this PR apply to?

- [x] Master (Default)
- [ ] Manager-4.2
- [ ] Manager-4.1
- [ ] Manager-4.0

# Links

Fixes https://github.com/SUSE/spacewalk/issues/17105

### CODE REVIEW

File: modules/specialized-guides/pages/salt/salt-ssh.adoc [hunk 1]
```diff
@@ -27,8 +27,15 @@ Use this method for in-place firewall environments that block HTTP or HTTPS conn
...
-{productname} keeps these details in the database and makes them available to Salt by generating a temporary roster file for each Salt SSH call.
-The location of the temporary roster file is supplied to `salt-ssh` using the `--roster-file= option`.
+{productname} keeps these details in the database and makes them available to Salt by using `uyuni` roster module or by generating a temporary roster file on bootstrapping new clients with {webui} using Salt SSH call.
+The location of the temporary roster file is supplied to `salt-ssh` using the `--roster-file=` option.
+For registered clients `--roster=uyuni` is used instead.
+
+[IMPORTANT]
+====
+It is not recommended to use [command]``salt-ssh`` as the call will be processed with current user permissions and in case of calling it with `root` user can cause permission issues for the files on processing next calls with `salt-api`.
+Please use [command]``mgr-salt-ssh`` instead. It is changing effective user to `salt` on each call. If you want to use it with a user other than `root`, the user should have the permission to change effective user to `salt`.
```
Comment (keichwa) [inline modules/specialized-guides/pages/salt/salt-ssh.adoc:37]:
```suggestion
Use [command]``mgr-salt-ssh`` instead because it is changing effective user to `salt` on each call.
If you want to use it with a user other than `root`, the user should have the permission to change effective user to `salt`.
```

File: modules/specialized-guides/pages/salt/salt-ssh.adoc [hunk 2]
```diff
@@ -178,6 +187,8 @@ The roster content contains:
 * `ssh_options`: other SSH options:
 ** `ProxyCommand` if the client connects through a proxy
 * `timeout`: defaults to 180 seconds
+* `ssh_pre_flight`: The path to the pre flight shell script (default: /usr/share/susemanager/salt-ssh/preflight.sh)
```
Comment (keichwa) [inline modules/specialized-guides/pages/salt/salt-ssh.adoc:190]:
```suggestion
* `ssh_pre_flight`: The path to the pre flight shell script (default: [path]``/usr/share/susemanager/salt-ssh/preflight.sh``)
```

File: modules/specialized-guides/pages/salt/salt-ssh.adoc [hunk 3]
```diff
@@ -27,8 +27,15 @@ Use this method for in-place firewall environments that block HTTP or HTTPS conn
 As with all Salt calls, {productname} invokes `salt-ssh` via the `salt-api`.
 
 Salt SSH relies on a roster to obtain details such as hostname, ports, and the SSH parameters of a client.
-{productname} keeps these details in the database and makes them available to Salt by generating a temporary roster file for each Salt SSH call.
-The location of the temporary roster file is supplied to `salt-ssh` using the `--roster-file= option`.
+{productname} keeps these details in the database and makes them available to Salt by using `uyuni` roster module or by generating a temporary roster file on bootstrapping new clients with {webui} using Salt SSH call.
+The location of the temporary roster file is supplied to `salt-ssh` using the `--roster-file=` option.
+For registered clients `--roster=uyuni` is used instead.
```
[the remark this answers is not in the document]
Comment (vzhestkov) [reply modules/specialized-guides/pages/salt/salt-ssh.adoc:32]: Actually the same as above. No `--` needed.

File: modules/specialized-guides/pages/salt/salt-ssh.adoc [hunk 4]
```diff
@@ -27,8 +27,15 @@ Use this method for in-place firewall environments that block HTTP or HTTPS conn
 As with all Salt calls, {productname} invokes `salt-ssh` via the `salt-api`.
 
 Salt SSH relies on a roster to obtain details such as hostname, ports, and the SSH parameters of a client.
-{productname} keeps these details in the database and makes them available to Salt by generating a temporary roster file for each Salt SSH call.
-The location of the temporary roster file is supplied to `salt-ssh` using the `--roster-file= option`.
+{productname} keeps these details in the database and makes them available to Salt by using `uyuni` roster module or by generating a temporary roster file on bootstrapping new clients with {webui} using Salt SSH call.
+The location of the temporary roster file is supplied to `salt-ssh` using the `--roster-file=` option.
```
[the remark this answers is not in the document]
Comment (keichwa) [reply modules/specialized-guides/pages/salt/salt-ssh.adoc]: ok, your explanation make sense (of course!).  In the long run, we should write "Salt SSH" when ever the subsystem is meant.  But this can come later, separately.  I think we now can merge your PR.  I unfortunately forgot again whether you have merge rights or not.  I can merge it for you.

File: modules/specialized-guides/pages/salt/salt-ssh.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (keichwa) [reply modules/specialized-guides/pages/salt/salt-ssh.adoc]: ok

**Review by keichwa — APPROVED** [review]
Thank you for also detecting additional typos such as wrong braces!

**Comment by vzhestkov** [comment]
@keichwa we've discussed the placement of one of the articles and I'm going to move it to the different section and extend slightly, so the PR not finished yet. I'm day off today, will ping you tomorrow as I have some questions regarding `l10n`.

**Comment by keichwa** [comment]
Victor Zhestkov ***@***.***> writes:

> @keichwa we've discussed the placement of one of the articles and I'm
> going to move it to the different section and extend slightly, so the
> PR not finished yet. I'm day off today, will ping you tomorrow as I
> have some questions regarding `l10n`.

Yes, also fine with me!  I did not want to urge you!  Some people get
nervous if I make too many change proposals ;-)

-- 
Karl Eichwalder

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc [hunk 5]
```diff
@@ -0,0 +1,81 @@
+[[contact-methods-salt]]
+= Salt Bundle
+
+== What is Salt Bundle?
+
+The Salt Bundle is a single binary package containing Salt Minion, Python 3, required Python modules and libraries.
+
+As the Salt Bundle is shipped with Python 3 and all the requirements for Salt. The Bundle is not using the Python installed on the system, so it can be installed on the client which doesn't meet the requirements for Salt version.
```
Comment (meaksh) [inline modules/client-configuration/pages/contact-methods-saltbundle.adoc:8]: I noticed something weird and just rephrase this as:

```suggestion
As the Salt Bundle is shipped with Python 3 and all the requirements for Salt to run, the Salt Bundle is not using the Python version coming from the system, so it can be installed on clients which doesn't meet the requirements for a given Salt version.
```

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+* Execute [command]``mgr-bootstrap`` with  [option]``--no-bundle`` option.
+* Set [literal]``AVOID_VENV_SALT_MINION`` to [literal]``1`` in the generated bootstrap script.
+* For bootstrap state set the [literal]``mgr_avoid_venv_salt_minion`` pillar to  [literal]``True``.
+
+== Salt SSH with the Salt Bundle
+
+The Salt Bundle is also used to handle Salt SSH session on the client side.
+
+The pre flight script is deploying the Salt Bundle on the client without installing the package. Original `salt-thin` method is not used in this case.
```
Comment (meaksh) [inline modules/client-configuration/pages/contact-methods-saltbundle.adoc:56]: Another suggestion: 

```suggestion
A pre flight script is executed as part of the Salt SSH execution which deploys the code from the Salt Bundle on the target client without installing the package. The original `salt-thin` method is not used in this case.
```

Maybe we could also mention here the paths where bundle gets deployed. Not sure.

**Review by meaksh — COMMENTED** [review]
Thanks for the text @vzhestkov !

I made some initial comments

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+Salt API is calling Python 3 deployed with the Bundle to handle Salt SSH session on the client, so Salt SSH (including bootstrapping with {webui}) is not dependant on Python installed on the system.
```
[the remark this answers is not in the document]
Comment (vzhestkov) [reply modules/client-configuration/pages/contact-methods-saltbundle.adoc:58]: We can avoid mentioning Salt API for example the following way:
```
The Python 3 deployed with the Bundle is used to handle Salt SSH session on the client, so Salt SSH (including bootstrapping with {webui}) is not dependant on Python installed on the system.
```
Is it better?

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc (hunk 6 above)
Comment (vzhestkov) [reply modules/client-configuration/pages/contact-methods-saltbundle.adoc:56]: Not sure if it makes any sense to the end users to know where it is placed, in general it's quite easy to find if they are really interested.

**Review by agraul — COMMENTED** [review]
Who is the target audience? Should this be for users who need to know how to use Salt Bundle and what its core components are? Or do we target developers who need to know technical details?

**Comment by vzhestkov** [comment]
The level of technical details is very different from one article to another. I've tried to keep balance and not to go too deep in the details.

**Review by 0rnela — APPROVED** [review]
Please add the change from @meaksh comment, the rest LGTM.

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+[IMPORTANT]
+====
+The bootstrap repository must be created before bootstrapping the client with {webui}. The pre flight script is detecting the OS on the client and using the bootstrap repository relevant for the OS version.
+For more information, see xref:client-configuration:bootstrap-repository.adoc#_prepare_to_create_a_bootstrap_repository[Prepare to Create a Bootstrap Repository]
+====
+
+[NOTE]
+====
+Using `salt-thin` can be enabled as a fallback method, but it requires Python 3 to be installed on the client.
+This method is not recommended and supported and can be used for development purposes only.
```
Comment (agraul) [inline modules/client-configuration/pages/contact-methods-saltbundle.adoc:69]:
```suggestion
This method is not recommended nor supported and exists for development purposes only.
```

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+----
+. Apply [literal]``util.mgr_switch_to_venv_minion`` with [literal]``mgr_purge_non_venv_salt`` set to [literal]``True`` to remove [package]``salt-minion`` and with [literal]``mgr_purge_non_venv_salt_files`` set to [literal]``True`` to remove all the files related to [package]``salt-minion``.
+This second step repeats the first step, and then removes the old configuration files and the now obsolete [package]``salt-minion`` package.
++
+----
+salt <minion_id> state.apply util.mgr_switch_to_venv_minion pillar='{"mgr_purge_non_venv_salt_files": True, "mgr_purge_non_venv_salt": True}'
+----
+
+On the other hand, it is also possible to avoid installing the Salt bundle and keep using [package]``salt-minion`` instead.
+In this case specify one of these options:
```
Comment (agraul) [inline modules/client-configuration/pages/contact-methods-saltbundle.adoc:46]:
```suggestion
In this case, specify one of these options:
```

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+* Execute [command]``mgr-bootstrap`` with  [option]``--no-bundle`` option.
+* Set [literal]``AVOID_VENV_SALT_MINION`` to [literal]``1`` in the generated bootstrap script.
+* For bootstrap state set the [literal]``mgr_avoid_venv_salt_minion`` pillar to  [literal]``True``.
+
+== Salt SSH with the Salt Bundle
+
+The Salt Bundle is also used when performing Salt SSH actions to clients.
+
+A pre flight script is executed as part of the Salt SSH execution which deploys the code from the Salt Bundle on the target client without installing the package. The original `salt-thin` method is not used in this case.
```
Comment (agraul) [inline modules/client-configuration/pages/contact-methods-saltbundle.adoc:56]: I would join this paragraph with the one below and reword it slightly.
```suggestion
A shell script deploys the Salt Bundle onto the target system without installing [package]``venv-salt-minion`` before any Salt command is executed. Since the Salt Bundle contains the whole Salt code base, no [literal]``salt-thin`` is deployed. Salt SSH (including bootstrapping using the {webui}) uses the Python 3 interpreter within the bundle. The target system does not need to have any other Python interpreter installed.
```

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc [hunk 11]
```diff
@@ -0,0 +1,81 @@
+[[contact-methods-salt]]
+= Salt Bundle
+
+== What is Salt Bundle?
+
+The Salt Bundle is a single binary package containing Salt Minion, Python 3, required Python modules and libraries.
+
+As the Salt Bundle is shipped with Python 3 and all the requirements for Salt to run, the Salt Bundle is not using the Python version coming from the system, so it can be installed on clients which doesn't meet the requirements for a given Salt version.
+
+It is also possible to install Salt Bundle on the system where Salt Minion is being used already to manage that client by the different Salt Master than the {productname} Server.
```
Comment (agraul) [inline modules/client-configuration/pages/contact-methods-saltbundle.adoc:10]:
```suggestion
It is also possible to use the Salt Bundle on systems that run a Salt Minion connected to a non-{poductname} Salt Master.
```

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+== Client Registration with Salt Bundle as a Minion
+
+The registration method with the Salt bundle is the recommended registration method as long as it covers your use case.
```
Comment (agraul) [inline modules/client-configuration/pages/contact-methods-saltbundle.adoc:14]: IMO that goes without saying.
```suggestion
The registration method with the Salt bundle is the recommended registration method.
```

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+This section explains the advantages and limitations of the current implementation.
+The Salt bundle is provided as the [package]``venv-salt-minion`` that consists of Salt, Python, and the Python modules Salt depends on.
```
Comment (agraul) [inline modules/client-configuration/pages/contact-methods-saltbundle.adoc:16]: Just for consistency with above
```suggestion
The Salt Bundle is provided as the [package]``venv-salt-minion`` that consists of Salt, Python 3, and the Python modules Salt depends on.
```

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+Bootstrapping with {webui} is using Salt bundle as well, so bootstrapping with {webui} is not Python dependant.
+Using the Salt bundle, it is no longer needed that the client provides the required Python version or any Python modules.
```
Comment (agraul) [inline modules/client-configuration/pages/contact-methods-saltbundle.adoc:18]:
```suggestion
Using the Salt Bundle, it is no longer needed that the client provides any Python interpreter or modules.
```

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+
+The Salt Bundle is also used when performing Salt SSH actions to clients.
+
+A pre flight script is executed as part of the Salt SSH execution which deploys the code from the Salt Bundle on the target client without installing the package. The original `salt-thin` method is not used in this case.
+
+Salt API is calling Python 3 deployed with the Bundle to handle Salt SSH session on the client, so Salt SSH (including bootstrapping with {webui}) is not dependant on Python installed on the system.
+
+[IMPORTANT]
+====
+The bootstrap repository must be created before bootstrapping the client with {webui}. The pre flight script is detecting the OS on the client and using the bootstrap repository relevant for the OS version.
```
Comment (agraul) [inline modules/client-configuration/pages/contact-methods-saltbundle.adoc:62]: I think we don't need to expose the "pre flight script" name to our users, I see it only as an implementation detail.
```suggestion
The bootstrap repository must be created before bootstrapping the client with {webui}. The Salt Bundle used by Salt SSH is taken from the bootstrap repository based on the detected target operating system.
```

File: modules/client-configuration/pages/registration-webui.adoc [hunk 16]
```diff
@@ -3,14 +3,18 @@
 
 Registering clients with the {productname} {webui} works for Salt clients only.
 
-If you are bootstrapping Salt clients using the {webui}, you need to ensure that the client system has Python installed before you begin.
-For Salt clients running {sles}{nbsp}12 or older, you also need the [package]``python-xml`` package.
-// FIXME: change Salt bundle, when implemented
-// If you use the Salt bundle, which now is the default, the needed Python environment will be provided automatically.
-// For more information, see xref:client-configuration:registration-methods-saltbundle.adoc[].
+If you are bootstrapping Salt clients using the {webui}, it is using xref:specialized-guides:salt/salt-ssh.adoc[Salt SSH] to execute the bootstrap process on the client.
+Salt bundle is used to handle Salt-SSH session on the client, so it is not Python dependant as Python is shipped with the bundle.
```
Comment (agraul) [inline modules/client-configuration/pages/registration-webui.adoc:7]:
```suggestion
Salt SSH uses the Salt Bundle and its included Python interpreter. Therefore, no other Python interpreter needs to be installed on the client.
```

File: modules/specialized-guides/pages/salt/salt-ssh.adoc [hunk 17]
```diff
@@ -27,8 +27,16 @@ Use this method for in-place firewall environments that block HTTP or HTTPS conn
...
-{productname} keeps these details in the database and makes them available to Salt by generating a temporary roster file for each Salt SSH call.
-The location of the temporary roster file is supplied to `salt-ssh` using the `--roster-file= option`.
+{productname} keeps these details in the database and makes them available to Salt by using `uyuni` roster module or by generating a temporary roster file on bootstrapping new clients with {webui} using Salt SSH call.
+The location of the temporary roster file is supplied to `salt-ssh` using the [option]``roster-file`` option.
+For the registered clients [option]``roster`` set to `uyuni` is used instead to get the roster from the database with `uyuni` salt roster module.
+
+[IMPORTANT]
+====
+It is not recommended to use [command]``salt-ssh`` as the call will be processed with current user permissions and in case of calling it with `root` user can cause permission issues for the files on processing next calls with `salt-api`.
+Use [command]``mgr-salt-ssh`` instead because it is changing effective user to `salt` on each call.
```
Comment (agraul) [inline modules/specialized-guides/pages/salt/salt-ssh.adoc:37]:
```suggestion
Use [command]``mgr-salt-ssh``, which changes the effective user to [literal]``salt``and avoids file permission issues.
```

File: modules/specialized-guides/pages/salt/salt-ssh.adoc [hunk 18]
```diff
@@ -39,7 +47,7 @@ Salt SSH supports both password and key authentication.
 
 Password authentication is used only when bootstrapping.
 During the bootstrap step the key of the server is not authorized on the client and therefore a password must be used for a connection to be made.
-The password is used transiently in a temporary roster file used for bootstrapping.
+The password is used transiently in a temporary roster file used for bootstrapping and the roster file is removed on finishing processing the event.
```
Comment (agraul) [inline modules/specialized-guides/pages/salt/salt-ssh.adoc:50]: IMO this change is not needed since "temporary roster file" implies that it's being removed afterwards.

**Review by agraul — CHANGES_REQUESTED** [review]
Overall I think this is very nice to have documented!

I left a few suggestions in-line. They range from tiny grammar fixes to some rephrasing that sounds better/clearer to me. Feel free to pick what you like and dismiss the rest.

The only thing that I would like to have changed are the "Salt bundle"->"Salt Bundle" changes (for consistency)

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc (hunk 11 above)
Comment (vzhestkov) [reply modules/client-configuration/pages/contact-methods-saltbundle.adoc:10]: Not really sure if `non-Uyuni` or `non-SUSE Manager` cold look nice there.
Maybe do it this way:
```suggestion
It is also possible to use the Salt Bundle on systems that run a Salt Minion connected to a Salt Master other than {poductname} Salt Master.
```

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc (hunk 12 above)
Comment (vzhestkov) [reply modules/client-configuration/pages/contact-methods-saltbundle.adoc:14]: I think it was the suggestion by @meaksh from the initial docs PR for Salt Bundle. Personally, I like the original more.

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc (hunk 11 above)
Comment (agraul) [reply modules/client-configuration/pages/contact-methods-saltbundle.adoc:10]: You're right, `non-SUSE Manager` looks bad. I tried to avoid repeating "Salt Master", but I agree that your version is better than mine.

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+Using the Salt bundle, it is no longer needed that the client provides the required Python version or any Python modules.
+
+If you bootstrap new clients, registration with the Salt bundle is the default method.
+You can switch existing clients to the Salt bundle method.
+If you switch, the [package]``salt-minion`` package and its Python version and the Python modules will stay installed.
+
+=== Switching from Salt Minion to Salt Bundle
+
+The Salt state [literal]``util.mgr_switch_to_venv_minion`` is available to switch from [package]``salt-minion`` to [package]``venv-salt-minion``.
+It is recommended to switch to [package]``venv-salt-minion`` in two steps to avoid trouble with shifting processes:
```
[the remark this answers is not in the document]
Comment (vzhestkov) [reply modules/client-configuration/pages/contact-methods-saltbundle.adoc:27]: This part was there since the initial PR with Salt Bundle documentation: https://www.uyuni-project.org/uyuni-docs/en/uyuni/client-configuration/registration-methods-saltbundle.html
Just in the other section. I haven't touched this part.

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+
+. Apply [literal]``util.mgr_switch_to_venv_minion`` with no pillar specified first.
+  This will result in the switch to [package]``venv-salt-minion`` with copying configuration files etc.
+  It will not clean up the original [package]``salt-minion`` configurations and its packages.
++
+----
+salt <minion_id> state.apply util.mgr_switch_to_venv_minion
+----
+. Apply [literal]``util.mgr_switch_to_venv_minion`` with [literal]``mgr_purge_non_venv_salt`` set to [literal]``True`` to remove [package]``salt-minion`` and with [literal]``mgr_purge_non_venv_salt_files`` set to [literal]``True`` to remove all the files related to [package]``salt-minion``.
+This second step repeats the first step, and then removes the old configuration files and the now obsolete [package]``salt-minion`` package.
```
[the remark this answers is not in the document]
Comment (vzhestkov) [reply modules/client-configuration/pages/contact-methods-saltbundle.adoc:39]: This can't be done in 1 step as it requires starting new minion service and stopping the old one. In general the second step is not repeating, but checking it the steps from first step were performed. Maybe need to rephrase it somehow.
```
The second step ensures that the first step passed and no changes required for it ....
```

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc (hunk 15 above)
Comment (vzhestkov) [reply modules/client-configuration/pages/contact-methods-saltbundle.adoc:62]: Not sure if it sounds more clear in passive than before.

File: modules/client-configuration/pages/registration-bootstrap.adoc [hunk 21]
```diff
@@ -121,6 +121,11 @@ ORG_GPG_KEY=
 
 . When you have finished, save the file, and repeat this procedure for the second bootstrap script.
 
+[NOTE]
+====
+By default bootstrap script will try to install [package]``venv-salt-minion`` for Salt clients if it's available in the bootstrap repository and [package]``salt-minion`` if there is no Salt bundle in the bootstrap repository.
+It is posible to avoid installing Salt bundle and keep using [package]``salt-minion`` if you need it for some reason. See xref:client-configuration:contact-methods-saltbundle.adoc[Salt Bundle] for details.
```
[the remark this answers is not in the document]
Comment (vzhestkov) [reply modules/client-configuration/pages/registration-bootstrap.adoc:127]: This sentance is mostly about the bootstrapping process so without mentioning installation it looks weird as we will continue use the salt-minion anyway.

File: modules/specialized-guides/pages/salt/salt-ssh.adoc [hunk 22]
```diff
@@ -27,8 +27,16 @@ Use this method for in-place firewall environments that block HTTP or HTTPS conn
 As with all Salt calls, {productname} invokes `salt-ssh` via the `salt-api`.
 
 Salt SSH relies on a roster to obtain details such as hostname, ports, and the SSH parameters of a client.
-{productname} keeps these details in the database and makes them available to Salt by generating a temporary roster file for each Salt SSH call.
-The location of the temporary roster file is supplied to `salt-ssh` using the `--roster-file= option`.
+{productname} keeps these details in the database and makes them available to Salt by using `uyuni` roster module or by generating a temporary roster file on bootstrapping new clients with {webui} using Salt SSH call.
+The location of the temporary roster file is supplied to `salt-ssh` using the [option]``roster-file`` option.
```
[the remark this answers is not in the document]
Comment (vzhestkov) [reply modules/specialized-guides/pages/salt/salt-ssh.adoc:31]: Actually `salt-ssh` is not a command in this case, in almost all the cases it's called in `salt-api` process.
So maybe the following is more correct (as it cold be in CLI as well as in salt-api event and the option has the same name for both cases):
```suggestion
The location of the temporary roster file is supplied to Salt SSH using the [option]``roster-file`` option.
```

File: modules/specialized-guides/pages/salt/salt-ssh.adoc (hunk 17 above)
Comment (vzhestkov) [reply modules/specialized-guides/pages/salt/salt-ssh.adoc:37]:
```suggestion
Use [command]``mgr-salt-ssh``, which changes the effective user to [literal]``salt``and avoids file permission issues. In addition it sets [option]``roster`` to [literal]``uyuni``, if neither [option]``roster`` nor [option]``roster-file`` specified, to allow using {productname} database as a source of roster data for the registered Salt SSH clients.
```

File: modules/specialized-guides/pages/salt/salt-ssh.adoc [hunk 23]
```diff
@@ -27,8 +27,16 @@ Use this method for in-place firewall environments that block HTTP or HTTPS conn
...
+For the registered clients [option]``roster`` set to `uyuni` is used instead to get the roster from the database with `uyuni` salt roster module.
+
+[IMPORTANT]
+====
+It is not recommended to use [command]``salt-ssh`` as the call will be processed with current user permissions and in case of calling it with `root` user can cause permission issues for the files on processing next calls with `salt-api`.
+Use [command]``mgr-salt-ssh`` instead because it is changing effective user to `salt` on each call.
+If you want to use it with a user other than `root`, the user should have the permission to change effective user to `salt`.
```
[the remark this answers is not in the document]
Comment (vzhestkov) [reply modules/specialized-guides/pages/salt/salt-ssh.adoc:38]: It's not actually related to the previous one. `mgr-salt-ssh` can't be used by any other user as it can't switch the effective user to `salt` in this case. It's mostly about this permission requirement.

File: modules/specialized-guides/pages/salt/salt-ssh.adoc (hunk 18 above)
Comment (vzhestkov) [reply modules/specialized-guides/pages/salt/salt-ssh.adoc:50]: Some temporary files could be left untouched on the system and from the security point of view it's better to mention it explicitly.

**Comment by vzhestkov** [comment]
@agraul @meaksh `contact-methods-saltssh.adoc` contains the following:
```
The server uses the [command]``salt-ssh`` tool to contact the clients at regular intervals, checking in and performing scheduled actions and events.
```
it's not true at all. Not sure how to rephrase it.
```
The server uses the Salt API to contact the clients at regular intervals, checking in and performing scheduled actions and events.
```
but in this case the sentance is related not only to Salt SSH managed systems, if case of replacing `Salt API` with `Salt SSH` the sentance looks redundant.

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc (hunk 19 above)
[the remark this answers is not in the document]
Comment (vzhestkov) [reply modules/client-configuration/pages/contact-methods-saltbundle.adoc:27]: Slightly rephased it and added extra section with the warning that the state apply could fail in case of running the second step with skipping the first.

File: modules/specialized-guides/pages/salt/salt-ssh.adoc (hunk 18 above)
Comment (agraul) [reply modules/specialized-guides/pages/salt/salt-ssh.adoc:50]: You're right that "temporary" only means "not forever" and could imply a file is kept for 1 day / week / year etc. If you'd rather be explicit, that's fine for me :+1:

File: modules/client-configuration/pages/registration-bootstrap.adoc (hunk 21 above)
[the remark this answers is not in the document]
Comment (0rnela) [reply modules/client-configuration/pages/registration-bootstrap.adoc:127]: It is possible to keep using [package]``salt-minion`` instead of the Salt Bundle if needed.

[the remark this answers is not in the document]
Comment (vzhestkov) [reply modules/client-configuration/pages/registration-bootstrap.adoc:127]: In this case the meaning of the sentance distorted a bit. The full meaning of this sentance that it's possible to avoid using Salt Bundle and use salt-minion package as usual if needed. But the sentance makes sense onlt for bootstrapping process, for the systems which was registered already nothing is changing, such systems will keep using salt-minion anyway.

**Comment by agraul** [comment]
> @agraul @meaksh `contact-methods-saltssh.adoc` contains the following:
> 
> ```
> The server uses the [command]``salt-ssh`` tool to contact the clients at regular intervals, checking in and performing scheduled actions and events.
> ```
> 
> it's not true at all. Not sure how to rephrase it.
> 
> ```
> The server uses the Salt API to contact the clients at regular intervals, checking in and performing scheduled actions and events.
> ```
> 
> but in this case the sentance is related not only to Salt SSH managed systems, if case of replacing `Salt API` with `Salt SSH` the sentance looks redundant.

I would use "Salt SSH", "Salt API" is a detail that's not interesting for most readers. While it is a bit repetitive, IMO the sentence is different enough to be included.

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc [hunk 24]
```diff
@@ -0,0 +1,81 @@
...
+Set [option]``web.ssh_use_salt_thin`` to `true` in the [path]``/etc/rhn/rhn.conf`` configuration file.
+====
+
+== Using the Salt Bundle with the Salt Minion
+
+The Salt Bundle can be used with the Salt Minion managed by the Salt Master other than {productname} Server at the same time.
+
+[IMPORTANT]
+====
+To bootstrap the client with the Salt Minion managed by the Salt Master other than {productname} Server it is recommended to use [command]``mgr-bootstrap`` [option]``--force-bundle`` on generating bootstrap script, or set [option]``FORCE_VENV_SALT_MINION`` to `1` in the bootstrap script.
```
[the remark this answers is not in the document]
Comment (vzhestkov) [reply modules/client-configuration/pages/contact-methods-saltbundle.adoc:84]: I'll fix it just a bit different way.

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc (hunk 12 above)
Comment (agraul) [reply modules/client-configuration/pages/contact-methods-saltbundle.adoc]: If we both like the shorter one more, let's see what Pablo says :smiley: 
@meaksh how badly do you want to keep `as long as it covers your use case`?

File: modules/specialized-guides/pages/salt/salt-ssh.adoc (hunk 17 above)
Comment (agraul) [reply modules/specialized-guides/pages/salt/salt-ssh.adoc:37]: I thought that this "important" section should be kept short and direct. Everything you wrote is technically correct, I just wonder if it could distract from the message `use mgr-salt-ssh` due to its length.

File: modules/specialized-guides/pages/salt/salt-ssh.adoc (hunk 23 above)
[the remark this answers is not in the document]
Comment (agraul) [reply modules/specialized-guides/pages/salt/salt-ssh.adoc:38]: It wasn't clear to me what "it" referred to. Then I would replace "it" with  

    [command]``mgr-salt-ssh``

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc (hunk 12 above)
Comment (meaksh) [reply modules/client-configuration/pages/contact-methods-saltbundle.adoc]: I'm fine removing that. It was not actually me who wrote that. :smile:

File: modules/client-configuration/pages/contact-methods-saltbundle.adoc (hunk 15 above)
Comment (meaksh) [reply modules/client-configuration/pages/contact-methods-saltbundle.adoc:67]: I'm also in favor of not mentioning pre flight script, maybe we can put the sentence in a different way but without mentioning pre flight script.

File: modules/specialized-guides/pages/salt/salt-ssh.adoc (hunk 17 above)
Comment (meaksh) [reply modules/specialized-guides/pages/salt/salt-ssh.adoc:37]: I see @agraul point here. I would prefer something short and effective sentence, then we can go and add the technical detail in a differente sentence maybe.

**Comment by keichwa** [comment]
Alexander Graul ***@***.***> writes:

> You're right, `non-SUSE Manager` looks bad. I tried to avoid repeating
> "Salt Master", but I agree that your version is better than mine.

This is technical documentation, and it is important to use the
technical terms consistently.  Repeat the same term again and again.
Even repeat phrases and syntactical constructs.

BTW, in our world Salt clients are Salt clients, "minion" is jargon that
we were asked to avoid.

-- 
Karl Eichwalder

SUSE Software Solutions Germany GmbH, Geschäftsführer: Ivo Totev
HRB 36809, AG Nürnberg  Maxfeldstr. 5,  D-90409 Nürnberg
